tests/framework/clients/pool.go

```diff
@@ -114,16 +114,20 @@ func (p *PoolOperation) DeletePool(blockClient *BlockOperation, clusterInfo *cli
 	blockImagesList, _ := blockClient.ListImagesInPool(clusterInfo, poolName)
 	for _, blockImage := range blockImagesList {
 		logger.Infof("force deleting block image %q in pool %q", blockImage, poolName)
+		max := 10
 		// Wait and retry up to 10 times/seconds to delete RBD images
-		for i := 0; i < 10; i++ {
+		for i := 0; i < max; i++ {
 			err := blockClient.DeleteBlockImage(clusterInfo, blockImage)
 			if err != nil {
 				logger.Infof("failed deleting image %q from %q. %v", blockImage, poolName, err)
 				time.Sleep(2 * time.Second)
 			} else {
 				break
 			}
-			return fmt.Errorf("gave up waiting for image %q from %q to be deleted. %v", blockImage, poolName, err)
+			if i == max-1 {
```

Comment:
interesting, I wonder if this could help fix some random failures in the tests when deleting pools

Reply:
Why not just `continue` at L123 instead?

Reply:
@leseb  We can do that but in that case [L126](https://github.com/rook/rook/pull/6306/files/d994455c75df4c807fb98971e3d903f6e55f9b94#diff-43987f988bb7062864f08037d020f9f3L126) will unreachable, we'll exit every time with if or else.

Reply:
Oh right, still morning for me I guess :)

Reply:
@leseb may be the latest works better, actually with previous changes also same the same situation as above. Thanks, after your comment only I looked again.